datastream-server/src/main/java/com/linkedin/datastream/server/Coordinator.java

```diff
@@ -644,6 +649,20 @@ private void handleAssignmentChange(boolean isDatastreamUpdate) throws TimeoutEx
         .flatMap(Collection::stream)
         .collect(Collectors.toMap(DatastreamTask::getDatastreamTaskName, Function.identity())));
 
+    if ((totalTasks - submittedTasks) > 0) {
+      _log.warn("Failed to submit {} tasks from currentAssignment. Queueing onAssignmentChange event again", totalTasks - submittedTasks);
+      // Update the metric and queue the event only once for all the tasks that failed to be initialized
+      if (isDatastreamUpdate) {
```

Comment:
Qn: Why do you need a different code logic for this v/s TimeoutException, when the expectation is the same in both the cases.

``` if (isDatastreamUpdate) {
        _eventQueue.put(CoordinatorEvent.createHandleDatastreamChangeEvent());
      } else {
        _eventQueue.put(CoordinatorEvent.createHandleAssignmentChangeEvent());
      }
      throw e;
``` There is no direct exception known in this path, but you can create DatastreamRuntimeException(). If you extract the highlighted code into a method and call it, it may work for you.

Logically it is the same code, but different way of achieving the same thing.